app/knight.py

```diff
@@ -0,0 +1,35 @@
+from equiption.armour import Armour
+from equiption.potion import Potion
+from equiption.weapon import Weapon
+
+
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list = None,
+                 weapon: dict = None,
+                 potion: dict = None) -> None:
+
+        self.name = name
+
+        self.weapon = Weapon(weapon["name"], weapon["power"])
+        self.power = power + self.weapon.power
+
+        self.hp = hp
+
+        self.armour = Armour(armour)
+        self.protection = (self.armour.helmet_protection
+                           + self.armour.boots_protection
+                           + self.armour.breastplate_protection)
+
+        if potion is not None:
+            pot = Potion(potion)
```

Comment:
The `potion` parameter is expected to be a dictionary with an 'effect' key containing a dictionary of effects. Ensure that this structure is maintained when passing data to the `Potion` class constructor .